tests/behaviour/expectations/semantic.ts

```diff
@@ -155,27 +159,52 @@ function transcodeTest(
   contractDef: ContractDefinition,
   { signature, callData, expectations, failure }: ITestCalldata,
   compilerVersion: string,
+  ast: AST,
 ): Expect {
   if (signature === '' || signature === '()') {
     throw new InvalidTestError('Fallback functions are not supported');
   }
-  if (signature === 'constructor()') {
-    throw new NotYetSuportedTestCaseError('Constructors not supported in tests yet');
-  }
 
   const [functionName] = signature.split('(');
 
   // Find the function definition in the ast
-  const defs = Array.from(resolveAny(functionName, contractDef, compilerVersion, true)).filter(
-    (def) => {
-      if (def instanceof FunctionDefinition) {
-        return def.canonicalSignature(ABIEncoderVersion.V2) === signature;
-      } else if (def instanceof VariableDeclaration) {
-        return def.getterCanonicalSignature(ABIEncoderVersion.V2) === signature;
-      }
-      return false;
-    },
-  ) as (FunctionDefinition | VariableDeclaration)[];
+  let defs: (FunctionDefinition | VariableDeclaration)[];
+  if (functionName !== 'constructor') {
+    defs = Array.from(resolveAny(functionName, contractDef, compilerVersion, true)).filter(
+      (def) => {
+        if (def instanceof FunctionDefinition) {
+          return def.canonicalSignature(ABIEncoderVersion.V2) === signature;
+        } else if (def instanceof VariableDeclaration) {
+          return def.getterCanonicalSignature(ABIEncoderVersion.V2) === signature;
+        }
+        return false;
+      },
+    ) as (FunctionDefinition | VariableDeclaration)[];
+  } else {
+    if (contractDef.vConstructor === undefined) {
+      // Need to create a default constructor and its abi
+      abi.push({ name: '', inputs: [], outputs: [] });
+      defs = [
+        new FunctionDefinition(
+          ast.reserveId(),
+          '',
+          contractDef.scope,
+          FunctionKind.Constructor,
+          '',
+          false,
+          FunctionVisibility.Public,
+          FunctionStateMutability.NonPayable,
+          true,
+          new ParameterList(ast.reserveId(), '', []),
+          new ParameterList(ast.reserveId(), '', []),
+          [],
+        ),
```

Comment:
There is a function `createDefaultConstructor` in `src/passes/inheritanceInliner/constructorInheritance` which does this exact same thing, you could move it into `utils/nodeTemplates` and use it instead.